R/datasets.R

```diff
@@ -0,0 +1,627 @@
+#' data in this dataset.
+#' @export
+#' @md
+convert_to_dataset_with_parquet_files <- function(dataset) {
+  dataset$to_parquet_files()
+}
+
+#' Configure conversion to bool.
+#'
+#' @description
+#' Configure conversion to bool.
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_bool <- function() {
+  azureml$data$dataset_factory$DataType$to_bool()
+}
+
+#' Configure conversion to datetime.
+#'
+#' @description
+#' Configure conversion to datetime.
+#'
+#' @param formats Formats to try for datetime conversion. For example `%d-%m-%Y` for data in "day-month-year",
+#' and `%Y-%m-%dT%H:%M:%S.%f` for "combined date an time representation" according to ISO 8601.
+#' * %Y: Year with 4 digits
+#' * %y: Year with 2 digits
+#' * %m: Month in digits
+#' * %b: Month represented by its abbreviated name in 3 letters, like Aug
+#' * %B: Month represented by its full name, like August
+#' * %d: Day in digits
+#' * %H: Hour as represented in 24-hour clock time
+#' * %I: Hour as represented in 12-hour clock time
+#' * %M: Minute in 2 digits
+#' * %S: Second in 2 digits
+#' * %f: Microsecond
+#' * %p: AM/PM designator
+#' * %z: Timezone, for example: -0700
+#'
+#' Format specifiers will be inferred if not specified.
+#' Inference requires that the data source is accessible from current compute.
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_datetime <- function(formats = NULL) {
+  azureml$data$dataset_factory$DataType$to_datetime(formats)
+}
+
+#' Configure conversion to 53-bit double.
+#'
+#' @description
+#' Configure conversion to 53-bit double.
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_double <- function()	{
+  as.double(azureml$data$dataset_factory$DataType$to_float())
+}
+
+#' Configure conversion to 64-bit integer.
+#'
+#' @description
+#' Configure conversion to 64-bit integer.
+#'
+#' @return Converted DataType object.
+#' @export
+#' @md
+data_type_long <- function() {
+  as.integer(azureml$data$dataset_factory$DataType$to_float())
+}
+
+#' Defines options for how column headers are processed when reading data from files to create a dataset.
+#'
+#' @description
+#' Defines options for how column headers are processed when reading data from files to create a dataset.
+#' These enumeration values are used in the Dataset class method.
+#'
+#' @param option A string describing how column headers are to be processed
+#' * NO_HEADERS No column headers are read
+#' * ONLY_FIRST_FILE_HAS_HEADERS Read headers only from first row of first file, everything else is data.
+#' * COMBINE_ALL_FILES_HEADERS Read headers from first row of each file, combining identically named columns.
+#' * ALL_FILES_HAVE_SAME_HEADERS Read headers from first row of first file, drops first row from other files.
+#' @return The PromoteHeadersBehavior object.
+#' @export
+#' @md
+promote_headers_behavior <- function(option) {
+  azureml$data$dataset_type_definitions$PromoteHeadersBehavior(option)
```

Comment:
this won't work.

Reply:
Why not? This is the way our enums are implemented in hyperdrive.R (see primary_metric_goal).